parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -264,10 +264,10 @@ if ! [[ ${API_SERVER_NAME} =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
             VALIDATION_ERR=$ERR_K8S_API_SERVER_DNS_LOOKUP_FAIL
         fi
     else
-        logs_to_events "AKS.CSE.apiserverNC" "retrycmd_if_failure ${API_SERVER_CONN_RETRIES} 1 10 nc -vz ${API_SERVER_NAME} 443 || time nc -vz ${API_SERVER_NAME} 443 || VALIDATION_ERR=$ERR_K8S_API_SERVER_CONN_FAIL"
+        logs_to_events "AKS.CSE.apiserverNC" "retrycmd_if_failure ${API_SERVER_CONN_RETRIES} 1 10 nc -vz ${API_SERVER_NAME} 443 || time nc -vz ${API_SERVER_NAME} 443" || VALIDATION_ERR=$ERR_K8S_API_SERVER_CONN_FAIL
```

Comment:
I think this actually was right the first time, I think now if the cmd fails, the logs_to_events func will return success, so it won't exit for sure

if you can just do a sanity check in local bash like
```bash
function logs_to_events() { ... }
logs_to_events "false || exit 1"
echo "made it"
````
and make sure it actually kills the shell (i.e., never reaches "made it"), that's fine for me.